r2/r2/lib/media.py

```diff
@@ -86,6 +303,36 @@ def str_to_image(s):
     image = Image.open(s)
     return image
 
+def svg_size(image):
+    """calculate the size of an SVG image"""
+    image = BeautifulSoup.BeautifulSoup(cStringIO.StringIO(image))
+    try:
+        dimensions = [image.svg['width'], image.svg['height']]
+
+        # Borrowed and edited from github.com/Zverik/svg-resize
+        for i in (0,1):
+            parts = re.match(r'^\s*(-?\d+(?:\.\d+)?)\s*(px|in|cm|mm|pt|pc|%)?', dimensions[i])
+            num = float(parts.group(1))
+            units = parts.group(2)
+            
+            if units == 'pt':
+                dimensions[i] = num * 1.25
+            elif units == 'pc':
+                dimensions[i] = num * 15.0
+            elif units == 'in':
+                dimensions[i] = num * 90.0
+            elif units == 'mm':
+                dimensions[i] = num * 3.543307
+            elif units == 'cm':
+                dimensions[i] = num * 35.43307
+            elif units == '%':
+                dimensions[i] = -num / 100.0
+            else:
+                dimensions[i] = num
+    except KeyError:
+        dimensions = [500, 500]
```

Comment:
The [500,500] is there because it's more complicated to actually run through each tag and calculate the overall width and height. It's completely arbitrary.

For the rest of that expression, it looks like the values are working for a screen at 90 ppi, although I did admittedly borrow these from someone else's script. The chances of someone using a unit other than 'px' is very low, though.

I think it's probably easier, rather than trying to implement every possible use-case, to make the user do the work here. All of the big SVG generators (Sketch, Inkscape, Illustrator) will specify width/height in px, so if there is an issue like an unspecified height, the user has probably crafted their tags from scratch.